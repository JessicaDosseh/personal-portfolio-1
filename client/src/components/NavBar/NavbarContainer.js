import React from "react";
import styled from "styled-components";
import NavbarMain from "./NavbarMain";
import PropTypes from "prop-types";

const NavbarContainerDiv = styled.div`
  width: 100%;
  max-width: 1200px;
  color: white;
  margin: 0 auto;
`;

const NavbarContainer = props => {
  return (
    <NavbarContainerDiv darkmode={props.darkmode}>
      <NavbarMain
        darkmode={props.darkmode}
        setDarkmode={props.setDarkmode}
        history={props.history}
      />
    </NavbarContainerDiv>
  );
};

NavbarContainer.propTypes = {
  darkmode: PropTypes.bool.isRequired
};

export default NavbarContainer;
