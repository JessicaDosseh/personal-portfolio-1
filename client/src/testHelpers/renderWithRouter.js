import React from "react";
import { Router } from "react-router-dom";
import { render } from "react-testing-library";
import { createMemoryHistory } from "history";
import { StateProvider } from "react-conflux";
import { darkContext } from "../store/contexts";
import { globalReducer } from "../store/reducers/globalReducer";

const renderWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...render(
      <StateProvider reducer={globalReducer} stateContext={darkContext}>
        <Router history={history}>{ui}</Router>
      </StateProvider>
    ),
    history
  };
};

export default renderWithRouter;
