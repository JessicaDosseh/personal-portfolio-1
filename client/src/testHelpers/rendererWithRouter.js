import React from "react";
import { Router } from "react-router-dom";
import renderer from "react-test-renderer";
import { createMemoryHistory } from "history";
import { StateProvider } from "react-conflux";
import { darkContext } from "../store/contexts";
import { globalReducer } from "../store/reducers/globalReducer";

const rendererWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...renderer.create(
      <StateProvider reducer={globalReducer} stateContext={darkContext}>
        <Router history={history}>{ui}</Router>
      </StateProvider>
    ),
    history
  };
};

export default rendererWithRouter;
