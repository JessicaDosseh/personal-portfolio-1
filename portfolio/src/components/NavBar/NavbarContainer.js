import React from "react";
import styled from "styled-components";

import NavbarMain from "./NavbarMain";

const NavbarContainerDiv = styled.div`
  width: 100%;
  color: white;
`;

const NavbarContainer = props => {
  return (
    <NavbarContainerDiv darkmode={props.darkmode}>
      <NavbarMain darkmode={props.darkmode} setDarkmode={props.setDarkmode} />
    </NavbarContainerDiv>
  );
};

export default NavbarContainer;
