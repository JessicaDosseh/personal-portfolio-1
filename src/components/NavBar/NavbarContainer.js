import React from "react";
import styled from "styled-components";

import NavbarMain from "./NavbarMain";

const NavbarContainerDiv = styled.div`
  width: 100%;
  max-width: 1200px;
  color: white;
  margin: 0 auto;
`;

const NavbarContainer = props => {
  return (
    <NavbarContainerDiv darkmode={props.darkmode}>
      <NavbarMain darkmode={props.darkmode} setDarkmode={props.setDarkmode} />
    </NavbarContainerDiv>
  );
};

export default NavbarContainer;
