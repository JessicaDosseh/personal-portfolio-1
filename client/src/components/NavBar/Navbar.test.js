import React from "react";
import { fireEvent } from "react-testing-library";
import "react-testing-library/cleanup-after-each";
import "jest-dom/extend-expect";
import { renderWithRouter, rendererWithRouter } from "../../testHelpers";

import NavbarContainer from "./NavbarContainer";

describe("<NavbarContainer />", () => {
  it("renders the Navbar component without crashing", () => {
    const mock = jest.fn();
    const helper = renderWithRouter(
      <NavbarContainer
        darkmode={false}
        setDarkmode={mock}
        history={{ location: { pathname: "/" } }}
      />
    );
  });

  it("matches the snapshot of NavbarContainer component", () => {
    const mock = jest.fn();
    const tree = rendererWithRouter(
      <NavbarContainer
        darkmode={false}
        setDarkmode={mock}
        history={{ location: { pathname: "/" } }}
      />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("should render the navbar text to the screen", () => {
    const mock = jest.fn();
    const { getByText } = renderWithRouter(
      <NavbarContainer
        darkmode={false}
        setDarkmode={mock}
        history={{ location: { pathname: "/" } }}
      />
    );
    expect(getByText(/Work/i)).toBeDefined();
    expect(getByText(/Contact/i)).toBeDefined();
  });
});
