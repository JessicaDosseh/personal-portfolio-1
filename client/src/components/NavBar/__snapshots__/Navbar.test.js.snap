// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavbarContainer /> matches the snapshot of NavbarContainer component 1`] = `
<div
  className="sc-ifAKCX iSFgrr"
>
  <div
    className="sc-bwzfXH hyjEOv"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <div
        className="sc-bdVaJa istZpN"
      >
        <div>
          <p>
            N
          </p>
        </div>
      </div>
    </a>
    <nav
      className="sc-htpNat jYdygF"
    >
      <div
        className="sc-bxivhb NNymt"
        data-testid="darkmodeTab"
        onClick={[Function]}
      >
        <div>
          <img
            alt="Dark mode logo"
            src="moon.svg"
          />
        </div>
      </div>
      <a
        href="#featured-work"
      >
        Work
      </a>
      <a
        aria-current={null}
        href="/contact"
        onClick={[Function]}
      >
        Contact
      </a>
    </nav>
  </div>
</div>
`;
